src/core/utils/tree.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export function isSubtree(leafPath: string, rootPath: string, delimiter = '.') {
```

Comment:
__issue/suggestion:__ I believe there was a discussion (that I agree with) about killing this `utils` folder entirely  at some point otherwise we'll end up with a catch-all folder for stuff that is ^kind of^ universally useful, but in fact it's not (especially since the second place where it's used is legacy that we should drop eventually).

So I'd vote for renaming generic `isSubtree` to something more narrowly specialized like `isConfigPath....blah-blah` and moving it to `config` "domain" instead (e.g. be a part of `src/core/server/config/config_service.ts` exports, but not necessarily). 

As a side benefit it'd allow us to drop configurable `delimiter` parameter and throw if either `leafPath` or `rootPath` is empty string (potentially relying on existing `isConfigPath(value: unknown)`, but it's out of scope) to catch possible hard-to-debug bugs early.

Reply:
yes, I did it at the first iteration and then found that I need it in [legacy platform as well](https://github.com/elastic/kibana/pull/36559/files#diff-fe1aaef8bd7a61991700deaee1c14cba). I found it weird to import from config service deeply, although it doesn't hurt much, so will do